webpack/src/raygun-webpack-plugin.ts

```diff
@@ -75,29 +76,53 @@ export class RaygunWebpackPlugin {
                 let body = '';
                 res.on('data', (chunk) => body += chunk);
                 res.on('end', () => {
+                    // Ensure statusCode is defined and check the range
                     if (typeof res.statusCode === 'number' && res.statusCode >= 200 && res.statusCode < 300) {
                         resolve();
                     } else {
-                        const errorMessage = `Failed to upload source map: HTTP status code ${res.statusCode}`;
-                        console.error(errorMessage); // Print the error message if non-2xx response
+                        const errorMessage = `Failed to upload source map: HTTP status code ${res.statusCode || 'undefined'}`;
+                        console.error(errorMessage);
                         reject(new Error(errorMessage));
                     }
                 });
             });
+
             req.on('error', (error) => {
-                reject(error);
                 console.error(`Request error: ${error}`);
+                reject(error);
             });
 
             req.write(postData);
             req.end();
         });
     }
+
     private getFileNameFromPath(filePath: string) {
         // Normalize the path to use a consistent separator
         const normalizedPath = filePath.replace(/\\/g, '/');
         // Split the path by the directory separator and get the last element
         const parts = normalizedPath.split('/');
         return parts.pop(); // Extracts and returns the last segment (the file name)
     }
+
+    private delay(ms: number): Promise<void> {
+        return new Promise(resolve => setTimeout(resolve, ms));
+    }
+
+    // Function to attempt an operation with retries and exponential backoff
+    private async retryOperation<T>(operation: () => Promise<T>, retries: number, delayLength: number): Promise<boolean> {
```

Comment:
you have a utils file, would it make sense for these to go in there?
I can also see the argument for keeping these in the file for simplicity so was just interested to here your thoughts on this